### PR TITLE
Improve error handling

### LINKED ISSUE #326: Precondition failed - login failure
There seems to a lot of login issues lately. Everything worked out just fine until this morning 11 a.m. CET. Integration fails to log in to API.

Yes, I have updated the Terms and Services and it did work ok for a while. Tried to log in to Skoda Store and also to skoda.vwgroup.io. I do get in and all the permissions are fine. Myskoda app works otherwise fine and is able to update car status (Skoda Enyaq 2023), but if I want to check my account in the app, it raises an error "Unexpected error, try again later".

This is what I see in integration logs. Sorry that some parts are in Finnish, but I believe you get the idea.

Tämä virhe on peräisin mukautetusta integroinnista.

`
Lokikirjaaja: custom_components.myskoda
Lähde: custom_components/myskoda/__init__.py:55
integraatio: MySkoda (dokumentaatio, ongelmat)
Ensimmäistä kertaa ilmennyt: 17.25.10 (5 tapahtumat)
Viimeksi kirjattu: 18.05.03

Login with MySkoda failed for an unknown reason.
Traceback (most recent call last):
  File "/config/custom_components/myskoda/__init__.py", line 55, in async_setup_entry
    await myskoda.connect(config.data["email"], config.data["password"])
  File "/usr/local/lib/python3.12/site-packages/myskoda/myskoda.py", line 124, in connect
    user = await self.get_user()
           ^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/myskoda/myskoda.py", line 296, in get_user
    return (await self.rest_api.get_user(anonymize=anonymize)).result
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/myskoda/rest_api.py", line 239, in get_user
    data=await self._make_get_request(url),
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/myskoda/rest_api.py", line 96, in _make_get_request
    return await self._make_request(url=url, method="GET")
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/myskoda/rest_api.py", line 86, in _make_request
    response.raise_for_status()
  File "/usr/local/lib/python3.12/site-packages/aiohttp/client_reqrep.py", line 1158, in raise_for_status
    raise ClientResponseError(
aiohttp.client_exceptions.ClientResponseError: 412, message='Precondition Failed', url='https://mysmob.api.connect.skoda-auto.cz/api/v1/users'
`

`Lokikirjaaja: myskoda.rest_api
Lähde: /usr/local/lib/python3.12/site-packages/myskoda/rest_api.py:92
Ensimmäistä kertaa ilmennyt: 17.25.10 (5 tapahtumat)
Viimeksi kirjattu: 18.05.03

Invalid status for GET request to /v1/users: 412
Traceback (most recent call last):
  File "/usr/local/lib/python3.12/site-packages/myskoda/rest_api.py", line 86, in _make_request
    response.raise_for_status()
  File "/usr/local/lib/python3.12/site-packages/aiohttp/client_reqrep.py", line 1158, in raise_for_status
    raise ClientResponseError(
aiohttp.client_exceptions.ClientResponseError: 412, message='Precondition Failed', url='https://mysmob.api.connect.skoda-auto.cz/api/v1/users'`

And at some point it gave this warning at the same exact time. Not sure if it's related.

`Lokikirjaaja: homeassistant
Lähde: runner.py:147
Ensimmäistä kertaa ilmennyt: 17.25.11 (4 tapahtumat)
Viimeksi kirjattu: 17.54.31

Error doing job: Unclosed client session (None)`

### PR DESCRIPTION
**Pull request by WebSpider**
- Improve generic error handling
- Handle HTTP error 412 by creating an issue (fixes #326)
- Made issue creation a bit more robust

Needs some real-world testing

### CODE REVIEW

**Comment by WebSpider** [comment]
This introduces a circular import. Fixing.

File: custom_components/myskoda/coordinator.py [hunk 1]
```diff
@@ -31,6 +31,8 @@
     DEFAULT_FETCH_INTERVAL_IN_MINUTES,
 )
 
+from . import handle_aiohttp_error
```
Comment (dvx76) [inline custom_components/myskoda/coordinator.py:34]: This "feels" 🙃 . Would it make sense to have this in an `error_handlers.py` instead?

File: custom_components/myskoda/utils.py [hunk 2]
```diff
@@ -1,10 +1,14 @@
+import logging
+
 from typing import Callable
 
 from homeassistant.helpers.entity_platform import AddEntitiesCallback
 
 from .coordinator import MySkodaDataUpdateCoordinator
 from .entity import MySkodaEntity
 
+_LOGGER = logging.getLogger(__name__)
```
Comment (dvx76) [inline custom_components/myskoda/utils.py:10]: Unused?

**Review by dvx76 — APPROVED** [review]
LGTM but as you mentioned needs real-world test(s)

File: custom_components/myskoda/coordinator.py (hunk 1 above)
Comment (WebSpider) [reply custom_components/myskoda/coordinator.py:34]: Yup. Will improve. Thanks! 👍

**Comment by WebSpider** [comment]
Doing some real world testing now.

**Pull request merged**